plugins/woocommerce/src/Internal/ProductAttributesLookup/Filterer.php

```diff
@@ -66,7 +66,8 @@ public function filter_by_attribute_post_clauses( array $args, \WP_Query $wp_que
 		// The extra derived table ("SELECT product_or_parent_id FROM") is needed for performance
 		// (causes the filtering subquery to be executed only once).
 		$clause_root = " {$wpdb->posts}.ID IN ( SELECT product_or_parent_id FROM (";
-		if ( 'yes' === get_option( 'woocommerce_hide_out_of_stock_items' ) ) {
+		$hide_out_of_stock = apply_filters('woocommerce_filterer_hide_out_of_stock', 'yes' === get_option( 'woocommerce_hide_out_of_stock_items' ) );
```

Comment:
The hook name is a bit too generic. What about simething like `woocommerce_product_filterer_hide_out_of_stock`?

Reply:
Makes sense. I think given the namespace and the class name `woocommerce_product_attributes_filterer_hide_out_of_stock` would be appropriate even if a bit long.